packages/editor-ui/src/components/Node/NodeCreator/composables/useActions.ts

```diff
@@ -216,6 +223,19 @@ export const useActions = () => {
 		return isCompatibleNode && isChatTriggerMissing;
 	}
 
+	// AI-226: Prepend LLM Chain node when adding a language model
+	function shouldPrependLLMChain(addedNodes: AddedNode[]): boolean {
+		const canvasHasAINodes = useCanvasStore().aiNodes.length > 0;
+		if (canvasHasAINodes) return false;
+
+		return addedNodes.some((node) => {
+			const nodeType = nodeTypesStore.getNodeType(node.type);
+			return Object.keys(nodeType?.codex?.subcategories ?? {}).includes(
```

Comment:
How do you mean?

Reply:
I mean something like `node.outputs.includes(NodeConnectionType.AiLanguageModel)`

Reply:
Because outputs are not always a simple array. It could also be an expression string, and to resolve that, we'd need to use `NodeHelpers.getNodeOutputs(workflow: Workflow, node: INode, nodeTypeData: INodeTypeDescription)`, which requires INode(UI Node), which we don't get before inserting the node into the canvas. But what would be the benefit? Both outputs and `codex.subcategories` and defined in Node's JSON and we already rely on subcategories a lot when rendering node creator items

Reply:
I see, it doesn't worth it then. My point was about "separation of concern", I (intuitively) expected that codex definitions affect **only** the position of the node in the catalog / appearance in search.